addons/l10n_hr/models/res_partner.py

```diff
@@ -0,0 +1,7 @@
+from odoo import models, fields
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    l10n_hr_personal_oib = fields.Char(string="Personal OIB")
```

Comment:
```suggestion
    l10n_hr_personal_oib = fields.Char(string="Personal OIB")
    
    @api.depends('vat', 'country_id')
    def _compute_company_registry(self):
        # OVERRIDE
        # If a croatian company has a VAT number then its company registry is its VAT Number (without country code).
        super()._compute_company_registry()
        for partner in self.filtered(lambda p: p._deduce_country_code() == 'HR' and p.vat):
            vat_country, vat_number = self._split_vat(partner.vat)
            if vat_country.isnumeric():
                vat_country = 'hr'
                vat_number = partner.vat
            if vat_country == 'hr' and self.simple_vat_check(vat_country, vat_number):
                partner.company_registry = vat_number
```

If a croatian company has a VAT number then its company registry is its VAT Number (without country code).